File: java/src/main/java/com/android/frame/http/model/BaseResponse.java
package com.android.frame.http.model;

import java.io.Serializable;

/**
 * Created by xuzhb on 2019/10/26
 * Desc:bean基类，非列表类型
 */
public class BaseResponse<T> implements Serializable {

    private int code;
    private String msg;
    private T data;

    public int getCode() {
        return code;
    }

    public void setCode(int code) {
        this.code = code;
    }

    public String getMsg() {
        return msg;
    }

    public void setMsg(String msg) {
        this.msg = msg;
    }

    public T getData() {
        return data;
    }

    public void setData(T data) {
        this.data = data;
    }

    public boolean isSuccess() {
        return this.code == 200;
    }

    public boolean isTokenOut() {  //登录失效
        return this.code == -111;  //这里假设code是-111
    }

    public String getMessage() {
        return msg;
    }

}

File: kotlin/src/main/java/com/android/frame/http/model/BaseResponse.kt
package com.android.frame.http.model

import java.io.Serializable

/**
 * Created by xuzhb on 2019/8/8
 * Desc:bean基类，非列表类型
 */
class BaseResponse<T>(
    val code: Int,
    val msg: String,
    val data: T?
) : Serializable {
    fun isSuccess(): Boolean = code == 200
    fun isTokenOut(): Boolean = code == -111  //登录失效，假设code是-111
    fun getMessage(): String = msg
}